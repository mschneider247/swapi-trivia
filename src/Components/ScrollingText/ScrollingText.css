article {
  width: 60%;
  height: 30%;
  overflow: hidden;
  margin-left: 20%;
}

.movie-title {
  color: #EEDB00
}

.scrolling-text {
  display: flex;
  justify-content: center;
  height: 300px;
  perspective: 100px;
  color: #EEDB00;
  font-family: 'Pathway Gothic One',
  sans-serif;
  font-size: 3vw;
  font-weight: 600;
  letter-spacing: 8px;
  line-height: 150%;
  text-align: center;
}

.crawl {
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 300s linear;
}

@keyframes crawl {
  0% {
    top: 200px;
    transform: rotateX(20deg) translateZ(120px);
  }

  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}