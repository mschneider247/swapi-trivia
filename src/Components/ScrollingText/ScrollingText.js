import React from 'react';
import './ScrollingText.css';
import PropTypes from 'prop-types';

const ScrollingText = ( props ) => {
  let romanNumerals = ['I', 'II', 'III', 'IV', 'V', 'VI', 'VII'];
  let romanEpisodeID = romanNumerals[parseInt(props.movieData.episode_id) - 1]

  return (
    <article>
      <section className="scrolling-text">
        <div className="crawl">
          <div className="title">
           <p>Episode {romanEpisodeID}</p>
            <h4 className="movie-title">{props.movieData.title}</h4>
          </div>
          <p>{props.movieData.opening_crawl}</p>
        </div>
      </section>
    </article>
  )
}

export default ScrollingText;

ScrollingText.propTypes = {
  moveData: PropTypes.object
}