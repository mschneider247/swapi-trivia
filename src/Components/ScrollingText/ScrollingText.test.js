import React from 'react';
import { shallow } from 'enzyme';
import ScrollingText from './ScrollingText';

describe('Scrolling Text', () => {
  
  beforeEach(() => {
  })
  
  it('Should match the snapshot', () => {
    const movieData = {episode_id: 4}
    const wrapper = shallow( <ScrollingText movieData={movieData}/> );
    expect(wrapper).toMatchSnapshot();
  })
})