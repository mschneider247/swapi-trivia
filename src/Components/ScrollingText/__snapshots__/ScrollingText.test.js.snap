// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrolling Text Should match the snapshot 1`] = `
<article>
  <section
    className="scrolling-text"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          Episode 
          4
        </p>
        <h4
          className="movie-title"
        />
      </div>
      <p />
    </div>
  </section>
</article>
`;
